### PR TITLE
fix(app-router): keep ownerless URL commits from releasing navigation snapshots

### PR DESCRIPTION
**Pull request by NathanDrake2406**
## What changed
Ownerless client URL sync commits no longer release an active App Router navigation render snapshot.

## Why
`commitClientNavigationState(undefined)` covered both ownerless URL syncs and superseded-transition cleanup. Hash-only navigation and patched `history.pushState` / `history.replaceState` could therefore consume a snapshot owned by an in-flight navigation. Once consumed, `usePathname()` and `useSearchParams()` stopped reading the pending render snapshot and could flicker back to stale committed URL state mid-transition.

## Approach
`commitClientNavigationState` now releases a snapshot only when called by a navigation owner (`navId`) or when a superseded App Router cleanup path explicitly opts into snapshot release. Ownerless URL sync paths still update committed pathname/search state and notify subscribers, but they do not balance `activateNavigationSnapshot()`.

The App Router superseded cleanup sites pass `{ releaseSnapshot: true }` while keeping `navId` undefined so they continue to avoid clearing `pendingPathname` owned by a newer navigation.

## Validation
- `vp test run tests/shims.test.ts -t "keeps pending render snapshot active"`
- `vp test run tests/app-browser-entry.test.ts`
- `vp test run tests/shims.test.ts`
- `vp check`

## Risks / follow-ups
The main risk is future superseded-transition cleanup code forgetting to opt into `releaseSnapshot`. The current App Router cleanup paths are explicit, and ownerless URL sync remains intentionally non-releasing.

## Next.js references
- [App Router pushState/replaceState restore path](https://github.com/vercel/next.js/blob/canary/packages/next/src/client/components/app-router.tsx#L303-L372)
- [App Router shallow routing hash navigation coverage](https://github.com/vercel/next.js/blob/canary/test/e2e/app-dir/shallow-routing/shallow-routing.test.ts#L450-L462)

### CODE REVIEW

**Comment by NathanDrake2406** [comment]
/bigbonk review this PR. Focus on blocking correctness issues and test gaps.

**Comment by NathanDrake2406** [comment]
/bigbonk review this PR. Focus on blocking correctness issues and whether the tests cover the real behavior and meaningful edge cases.

File: packages/vinext/src/shims/navigation.ts [hunk 1]
```diff
@@ -1016,19 +1020,23 @@ function withSuppressedUrlNotifications<T>(fn: () => T): T {
...
 
-  // Only decrement the snapshot counter if a snapshot was previously activated.
-  // Several code paths call commit without a prior activateNavigationSnapshot()
-  // — hash-only changes (navigateClientSide), Pages Router popstate, and
-  // patched history.pushState/replaceState — which legitimately have count == 0.
-  if (state.navigationSnapshotActiveCount > 0) {
+  // Only navigation-owned commits may release a render snapshot. Ownerless URL
+  // syncs still update committed pathname/search state, but must not consume
+  // the active snapshot for an in-flight App Router transition.
+  const shouldReleaseSnapshot = navId !== undefined || options?.releaseSnapshot === true;
```
Comment (james-elicx) [inline packages/vinext/src/shims/navigation.ts:1038]:
```suggestion
  const shouldReleaseSnapshot = navId !== undefined || options?.releaseSnapshot;
```

nit: shouldnt be possible to be a truthy value other than true due to the types
